x-pack/legacy/plugins/security/public/hacks/legacy.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { uiModules } from 'ui/modules';
+import { npSetup, npStart } from 'ui/new_platform';
+import routes from 'ui/routes';
+import { isSystemApiRequest } from '../../../../../../src/plugins/kibana_legacy/public';
+import { SecurityPluginSetup } from '../../../../../plugins/security/public';
+
+const securityPluginSetup = (npSetup.plugins as any).security as SecurityPluginSetup;
+if (securityPluginSetup) {
+  routes.when('/account', {
+    template: '<div />',
+    controller: () => npStart.core.application.navigateToApp('security'),
+  });
+
+  const getNextParameter = () => {
+    const { location } = window;
+    const next = encodeURIComponent(`${location.pathname}${location.search}${location.hash}`);
+    return `&next=${next}`;
+  };
+
+  const getProviderParameter = (tenant: string) => {
+    const key = `${tenant}/session_provider`;
+    const providerName = sessionStorage.getItem(key);
+    return providerName ? `&provider=${encodeURIComponent(providerName)}` : '';
+  };
+
+  const module = uiModules.get('security', []);
+  module.config(($httpProvider: ng.IHttpProvider) => {
+    $httpProvider.interceptors.push(($q, $window, Promise) => {
+      const isAnonymous = npSetup.core.http.anonymousPaths.isAnonymous(window.location.pathname);
+
+      function interceptorFactory(responseHandler: (response: ng.IHttpResponse<unknown>) => any) {
+        return function interceptor(response: ng.IHttpResponse<unknown>) {
+          // TODO: SHOULD WE CHECK THAT IT'S NOT ERROR RESPONSE (&& response.status !== 401)?
```

Comment:
__note:__ I think I'll just keep it as is and remove this TODO, as it [seems to be intentional/known](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/public/session/session_timeout.tsx#L117).

Reply:
> The linked catch clause is unrelated -- that's catching errors for outgoing session info requests.

I meant that this code seems to be called via `sessionTimeout.extend` we call here even if interceptor intercepted `401` response (assuming `warningToast` is shown). Unless I completely misread this code :slightly_smiling_face: